src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilterTest.java

```diff
@@ -400,6 +401,59 @@ public void testDecideById() throws Exception {
                 expected, removeSuppressed(expected, suppressed));
     }
 
+    /**
+     * This test is required to cover pitest mutation
+     * for reset of 'file' field in SuppressionXpathSingleFilter.
+     * This not possible to reproduce by natural execution of Checkstyle
+     * as config is never changed in runtime.
+     * Projects that use us by api can reproduce this case.
+     * We need to allow users to reset module property to default state.
+     *
+     * @throws Exception when there is problem to load Input file
+     */
+    @Test
+    public void testUpdateFilterSettingsInRunTime() throws Exception {
+        final File file = new File(getPath("InputSuppressionXpathSingleFilterComplexQuery.java"));
+
+        final SuppressionXpathSingleFilter filter = new SuppressionXpathSingleFilter();
+        filter.setChecks("MagicNumber");
+        filter.finishLocalSetup();
+
+        final Violation violation = new Violation(27, 21, TokenTypes.NUM_DOUBLE, "",
+                        "", null, null, null,
+                        MagicNumberCheck.class, null);
+
+        final FileContents fileContents =
+                new FileContents(new FileText(file, StandardCharsets.UTF_8.name()));
+
+        final TreeWalkerAuditEvent ev = new TreeWalkerAuditEvent(fileContents, file.getName(),
+                violation, JavaParser.parseFile(file, JavaParser.Options.WITHOUT_COMMENTS));
+
+        assertWithMessage("match ia expected as 'files' is defaulted")
+                .that(filter.accept(ev))
+                .isFalse();
+
+        // set non-default value for files
+        filter.setFiles(sanitizeWindowsPathForRegexp(file.getPath() + ".never.match"));
+        filter.finishLocalSetup();
+
+        assertWithMessage("no match is expected due to weird value in 'files'")
+                .that(filter.accept(ev))
+                .isTrue();
+
+        // reset files to default value of filter
+        filter.setFiles(null);
+        filter.finishLocalSetup();
+
+        assertWithMessage("match ia expected as 'files' is defaulted")
+                .that(filter.accept(ev))
+                .isFalse();
+    }
+
+    private static String sanitizeWindowsPathForRegexp(String value) {
```

Comment:
This has nothing to do with Windows. It is just linux was not smart enough to notice the issue.

Reply:
What name you suggest to use?
In Linux there is no `\` in path, so I tried by name of method explaining what am I targeting to address.

Reply:
> In Linux there is no \ in path

This is why linux did not find the issue. It is a sanitation issue and the only way to identify it was to give it something that wasn't regexp safe. The issue would have presented itself if the linux path had some things that weren't regexp safe.

> What name you suggest to use?

`sanitizePathForRegexp`

In addition, I recommend to just use `Pattern.quote`.
https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#quote-java.lang.String-

Reply:
Pattern.quote is used.